src/Utilities/hooks/useFetchBatched.js

```diff
@@ -36,3 +36,34 @@ const useFetchBatched = () => {
 };
 
 export default useFetchBatched;
+
+// Includes pageOffset
+export const useCustomFetchBatched = () => {
+  const { isResolving: isLoading, resolve } = usePromiseQueue();
+
+  return {
+    isLoading,
+    fetchBatched: (
+      fetchFunction,
+      total,
+      filter,
+      batchSize = 50,
+      pageOffset = 0
+    ) => {
+      const pages = Math.ceil(total / batchSize) || 1;
+
+      const results = resolve(
+        [...new Array(pages)].map(
+          // eslint-disable-next-line camelcase
+          (_, pageIdx) => () =>
+            fetchFunction(filter, {
+              page: pageOffset + pageIdx + 1,
+              per_page: batchSize,
+            })
+        )
+      );
+
+      return results;
+    },
```

Comment:
Maybe I'm missing the point of this, but could we not rename this as `pageOffsetFetchBatched` and include it here: https://github.com/RedHatInsights/insights-inventory-frontend/blob/2cedc36f31e0966b4ef869d0cf56d0f2fd448f77/src/Utilities/hooks/useFetchBatched.js#L34C1-L35

Then we could import it as `const { pageOffsetFetchBatched } = useFetchBatched();`

Reply:
+1 We should rather extend the existing hook, than create a new one that looks almost exactly the same just with one minor difference. 

On that note, the existing hook already has two very similar functions that it returns, which we should also refactor.

Reply:
Ok I'll extend the existing hook